### PR TITLE
Fix TargetFramework handling in a few places

### PR DESCRIPTION
**Pull request by dsplaisted**
- Treat TargetFramework as a possible alias
- Fix targeting pack matching to match on target platform identifier
- Update WindowsDesktop SDK warning text
- Other minor fixes

### CODE REVIEW

File: src/Tests/Microsoft.NET.Build.Tests/GivenFrameworkReferences.cs [hunk 1]
```diff
@@ -261,6 +261,46 @@ public void KnownFrameworkReferencesOnlyApplyToCorrectTargetFramework(string tar
...
+                                               new XAttribute("RuntimeFrameworkName", "NonExistentTestFrameworkReference"),
+                                               new XAttribute("DefaultRuntimeFrameworkVersion", "7.0"),
+                                               new XAttribute("LatestRuntimeFrameworkVersion", "7.0"),
+                                               new XAttribute("TargetingPackName", "NonExistentTestFrameworkReference"),
+                                               new XAttribute("TargetingPackVersion", "7.0")));
+                });
+
+            var buildCommand = new BuildCommand(testAsset);
+
+            //  The build should succeed because the fake KnownFrameworkReference should not match, and the SDK shouldn't try to download
```
Comment (joeloff) [inline src/Tests/Microsoft.NET.Build.Tests/GivenFrameworkReferences.cs:297]: Is there a warning if there's no match?

File: src/Tests/Microsoft.NET.Build.Tests/DesignTimeBuildTests.cs [hunk 2]
```diff
@@ -210,6 +262,7 @@ private static string[] GetDesignTimeMSBuildArgs()
                 "/t:CollectResolvedCompilationReferencesDesignTime;ResolveFrameworkReferences",
                 //  Set targeting pack folder to nonexistant folder so the project won't use installed targeting packs
                 "/p:NetCoreTargetingPackRoot=" + Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString()),
+                "/bl"
```
Comment (sfoslund) [inline src/Tests/Microsoft.NET.Build.Tests/DesignTimeBuildTests.cs:265]: Leftover from debugging?

Comment (dsplaisted) [reply src/Tests/Microsoft.NET.Build.Tests/DesignTimeBuildTests.cs]: Yes, thanks! :-)

**Review by wli3 — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks/ResolveToolPackagePaths.cs [hunk 3]
```diff
@@ -50,7 +52,10 @@ protected override void ExecuteCore()
                     Path.Combine(PublishDir,
                     relativePath));
                 var i = new TaskItem(fullpath);
-                i.SetMetadata("PackagePath", $"tools/{TargetFramework}/any/{GetDirectoryPathInRelativePath(relativePath)}");
+
+                var shortFrameworkName = NuGetFramework.Parse(TargetFrameworkMoniker).GetShortFolderName();
```
Comment (nkolev92) [inline src/Tasks/Microsoft.NET.Build.Tasks/ResolveToolPackagePaths.cs:56]: oops :D

**Review by nkolev92 — APPROVED** [review]
LGTM

File: src/Tests/Microsoft.NET.Build.Tests/GivenFrameworkReferences.cs (hunk 1 above)
Comment (dsplaisted) [reply src/Tests/Microsoft.NET.Build.Tests/GivenFrameworkReferences.cs:297]: There is a list of allowed versions and an error if you use something not on the list.

There's not an error if there's no targeting pack for a platform.  This is supposed to be the case for Windows 7.0.

**Pull request merged**